c++/src/kj/string.h

```diff
@@ -136,15 +145,17 @@ class StringPtr {
   // Overflowed floating numbers return inf.
 
 private:
-  inline constexpr StringPtr(ArrayPtr<const char> content): content(content) {}
+  inline explicit constexpr StringPtr(ArrayPtr<const char> content): content(content) {}
 
   ArrayPtr<const char> content;
 
   friend constexpr kj::StringPtr (::operator "" _kj)(const char* str, size_t n);
 };
 
+#if __cplusplus < 202000L
 inline bool operator==(const char* a, const StringPtr& b) { return b == a; }
```

Comment:
Note: this change does not just fix a compile-time error. It also fixes an infinite recursion bug if this overload is ever called by c++20 code.

Specifically, with Cap'n Proto 0.8.1 and earlier, if code is compiled against this header with `-std=c++20` and uses a `var == "literal"` or `"literal" == var` comparison where `var` is `StringPtr`, the code will either crash or loop infinitely without this fix.

If the same code is compiled with `-std=c++17` there is no problem because instead of this function calling itself recursively, it will implicitly convert `a` to a `StringPtr` call the StringPtr operator== method instead.

This problem was encountered at https://github.com/bitcoin/bitcoin/issues/33176 on ubuntu 22.04 which uses v0.8.0.